examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+# Copyright 2014-2016 The ODL development group
```

Comment:
Okay, sure. My view is that we're going to have at least two, probably three different frameworks for deformation (linearized, optimal information transport and possibly LDDMM).

- **Interpolation code:** This is a bit tricky since it is low-level functionality but on the other hand uses advanced features. For example, the kernel interpolation makes use of the Fourier transform, which makes it hard to place in the ecosystem. Most natural would be `discr_mappings` -- here one could think of a split into `sampling` and `interpolation` since it's going to be quite an amount of code -- but that would require great care to avoid circular dependencies.

- **Deformation code:** That part is not going to depend on very much other stuff and can be placed anywhere. However, it's also going to be a significant amount of code, so appending it to some existing module is not really an option IMO. Having a sub*package* is another thing, though. One could instead place it as a new module under `discr`. However, I'm not sure it really fits there.